src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -223,13 +223,12 @@ internal partial class Binder
                             var typeSyntax = typeConstraintSyntax.Type;
                             var typeSyntaxKind = typeSyntax.Kind();
 
-                            // For pointer types, don't report this error. It is already reported during binding typeSyntax below.
                             switch (typeSyntaxKind)
                             {
                                 case SyntaxKind.PredefinedType:
-                                case SyntaxKind.PointerType:
                                 case SyntaxKind.NullableType:
                                     break;
+                                case SyntaxKind.PointerType:
```

Comment:
>case SyntaxKind.PointerType: [](start = 32, length = 28)

I am not sure why are we splitting reporting invalid constraint types between this place and Binder.IsValidConstraintType. Moreover, this place does it based on the syntax, which would still leave a hole if we could have aliases for pointers and array types. If in the future we relax the alias restriction, we will get in trouble. Consider consolidating the checks to one place and performing it based on symbol.